src/cert_context.rs

```diff
@@ -130,6 +130,13 @@ impl CertContext {
         }
     }
 
+    /// Compares two certificates and returns true, if they are identical
+    pub fn is_equal(&self, other: &CertContext) -> io::Result<bool> {
```

Comment:
If all we're doing is comparing for equality we should be able to just compare the encoded bytes directly to avoid the Err case here.